File: apps/mobile/lib/features/agency_platform/agency_platform_constants.dart
/// BloomOra agency platform module keys — must match API `AGENCY_PLATFORM_MODULE_KEYS`.
class AgencyPlatformModules {
  AgencyPlatformModules._();

  static const dashboard = 'dashboard';
  static const clients = 'clients';
  static const providers = 'providers';
  static const agencies = 'agencies';
  static const serviceCoordination = 'service_coordination';
  static const scheduling = 'scheduling';
  static const sessionNotes = 'session_notes';
  static const documents = 'documents';
  static const billing = 'billing';
  static const payroll = 'payroll';
  static const referrals = 'referrals';
  static const outreach = 'outreach';
  static const reports = 'reports';
  static const integrations = 'integrations';
  static const adminSettings = 'admin_settings';
  static const auditLogs = 'audit_logs';

  static const all = [
    dashboard,
    clients,
    providers,
    agencies,
    serviceCoordination,
    scheduling,
    sessionNotes,
    documents,
    billing,
    payroll,
    referrals,
    outreach,
    reports,
    integrations,
    adminSettings,
    auditLogs,
  ];

  static const labels = <String, String>{
    dashboard: 'Dashboard',
    clients: 'Clients',
    providers: 'Providers',
    agencies: 'Agencies',
    serviceCoordination: 'Service Coordination',
    scheduling: 'Scheduling',
    sessionNotes: 'Session Notes',
    documents: 'Documents',
    billing: 'Billing',
    payroll: 'Payroll',
    referrals: 'Referrals',
    outreach: 'Outreach',
    reports: 'Reports',
    integrations: 'Integrations',
    adminSettings: 'Admin Settings',
    auditLogs: 'Audit Logs',
  };
}

/// Full client profile tabs for Phase 1 shell (content fills in over later phases).
class ClientProfileTabs {
  ClientProfileTabs._();

  static const overview = 'Overview';
  static const family = 'Family / Contacts';
  static const program = 'Program';
  static const services = 'Services';
  static const authorizations = 'Authorizations';
  static const assignedProviders = 'Providers';
  static const insurance = 'Insurance';
  static const medical = 'Medical';
  static const diagnosis = 'Diagnosis';
  static const prescriptions = 'Prescriptions / Referrals';
  static const attendance = 'Attendance';
  static const sessions = 'Sessions';
  static const documents = 'Documents';
  static const goals = 'Goals';
  static const progressReports = 'Progress Reports';
  static const meetings = 'Meetings / Reviews';
  static const billing = 'Billing';
  static const notes = 'Notes';
  static const auditLog = 'Audit Log';

  static const all = [
    overview,
    family,
    program,
    services,
    authorizations,
    assignedProviders,
    insurance,
    medical,
    diagnosis,
    prescriptions,
    attendance,
    sessions,
    documents,
    goals,
    progressReports,
    meetings,
    billing,
    notes,
    auditLog,
  ];
}

/// Provider profile tabs for Phase 1 shell.
class ProviderProfileTabs {
  ProviderProfileTabs._();

  static const overview = 'Overview';
  static const credentials = 'Credentials';
  static const compliance = 'Compliance';
  static const caseload = 'Caseload';
  static const services = 'Services';
  static const availability = 'Availability';
  static const payroll = 'Payroll';
  static const documents = 'Documents';
  static const portalAccess = 'Portal Access';
  static const auditLog = 'Audit Log';

  static const all = [
    overview,
    credentials,
    compliance,
    caseload,
    services,
    availability,
    payroll,
    documents,
    portalAccess,
    auditLog,
  ];
}

const bloomoraComplianceDisclaimer =
    'BloomOra is a technology and care-navigation platform. It does not provide '
    'medical diagnosis, determine program eligibility, or replace professional '
    'evaluation. Final decisions must follow applicable laws, payer rules, '
    'agency policies, and professional standards.';

File: api/src/agency-platform/agency-platform.constants.ts
export const AGENCY_PLATFORM_MODULE_KEYS = [
  'dashboard',
  'clients',
  'providers',
  'agencies',
  'service_coordination',
  'scheduling',
  'session_notes',
  'documents',
  'billing',
  'payroll',
  'referrals',
  'outreach',
  'reports',
  'integrations',
  'admin_settings',
  'audit_logs',
] as const;

export type AgencyPlatformModuleKey =
  (typeof AGENCY_PLATFORM_MODULE_KEYS)[number];

export const AGENCY_PLATFORM_MODULE_LABELS: Record<
  AgencyPlatformModuleKey,
  string
> = {
  dashboard: 'Dashboard',
  clients: 'Clients',
  providers: 'Providers',
  agencies: 'Agencies',
  service_coordination: 'Service Coordination',
  scheduling: 'Scheduling',
  session_notes: 'Session Notes',
  documents: 'Documents',
  billing: 'Billing',
  payroll: 'Payroll',
  referrals: 'Referrals',
  outreach: 'Outreach',
  reports: 'Reports',
  integrations: 'Integrations',
  admin_settings: 'Admin Settings',
  audit_logs: 'Audit Logs',
};

export const DEFAULT_ENABLED_MODULES: Record<
  AgencyPlatformModuleKey,
  boolean
> = {
  dashboard: true,
  clients: true,
  providers: true,
  agencies: true,
  service_coordination: true,
  scheduling: true,
  session_notes: true,
  documents: true,
  billing: true,
  payroll: false,
  referrals: true,
  outreach: false,
  reports: true,
  integrations: false,
  admin_settings: true,
  audit_logs: true,
};

export const PERMISSION_SCOPE_TYPES = [
  'ROLE',
  'DEPARTMENT',
  'PROGRAM',
  'USER',
] as const;

export type PermissionScopeType = (typeof PERMISSION_SCOPE_TYPES)[number];

export const BLOOMORA_COMPLIANCE_DISCLAIMER =
  'BloomOra is a technology and care-navigation platform. It does not provide medical diagnosis, determine program eligibility, or replace professional evaluation. Final decisions must follow applicable laws, payer rules, agency policies, and professional standards.';

export const INTEGRATION_CATALOG = [
  {
    key: 'edi_837',
    label: '837 Professional Claims',
    category: 'clearinghouse',
    description: 'Generate and submit professional billing files.',
  },
  {
    key: 'edi_835',
    label: '835 Remittance',
    category: 'clearinghouse',
    description: 'Process payer remittance and payment files.',
  },
  {
    key: 'edi_278',
    label: '278 Authorization',
    category: 'clearinghouse',
    description: 'Authorization request and response workflows.',
  },
  {
    key: 'ny_eis',
    label: 'NY EIS / EI-Hub',
    category: 'municipality',
    description: 'New York Early Intervention state systems.',
  },
  {
    key: 'promise',
    label: 'PROMISe (PA)',
    category: 'municipality',
    description: 'Pennsylvania PROMISe fiscal agent workflows.',
  },
  {
    key: 'tkids',
    label: 'TKIDS',
    category: 'municipality',
    description: 'Texas ECI TKIDS-compatible exports.',
  },
  {
    key: 'payroll_export',
    label: 'Payroll company export',
    category: 'payroll',
    description: 'CSV/API exports for external payroll vendors.',
  },
] as const;
